### PR TITLE
shared_ptr<ProjDataInfo> -> shared_ptr<const ProjDataInfo>

### PR DESCRIPTION
**Pull request by rijobro**
Internally use shared_ptr<const ProjDataInfo> where possible

### CODE REVIEW

**Comment by rijobro** [comment]
fixes #479

**Comment by KrisThielemans** [comment]
This is too much for me :-(

Instead, I'd like to see a PR that only changes .h files, and corresponding internal implementations. No "outside" code should be affected (I hope)

**Comment by rijobro** [comment]
don't think I can do more minimal than the second commit.

**Comment by rijobro** [comment]
> No "outside" code should be affected (I hope)

Outside code will be affected if e.g., `get_proj_data_info` was called, since the result will now be `const`, so the outside code needs to either clone or store it as a const, too.

File: src/include/stir/recon_buildblock/PoissonLogLikelihoodWithLinearKineticModelAndDynamicProjectionData.txx [hunk 1]
```diff
@@ -304,7 +304,7 @@ set_up_before_sensitivity(shared_ptr<TargetT > const& target_sptr)
       return Succeeded::no;
     }
 
-  shared_ptr<ProjDataInfo> proj_data_info_sptr(
+  const shared_ptr<ProjDataInfo> proj_data_info_sptr(
```
Comment (KrisThielemans) [inline src/include/stir/recon_buildblock/PoissonLogLikelihoodWithLinearKineticModelAndDynamicProjectionData.txx:307]: not really needed, and maybe somewhat confusing. (it's a const pointer to a non-const object), but as you prefer

**Comment by rijobro** [comment]
Ok to (squash) merge this? Travis is happy.

**Comment by KrisThielemans** [comment]
Let's remove the `Use with care` warnings in the proj-data classes (do a search in `include/stir` only, not in subdirs). Then squash merge

**Comment by KrisThielemans** [comment]
ah, you're faster than me! A tiny PR for that then I guess

**Comment by rijobro** [comment]
Did that in the last commit before merge, is this what you wanted? https://github.com/UCL/STIR/pull/480/commits/f92a502d315387ab425a1de93509cb83f7bb3d12

**Comment by KrisThielemans** [comment]
you're even faster than I think you are! great. thanks

**Pull request merged**